### DISCUSSION TITLE
How do I add EasyOCR

### DISCUSSION THREAD

**Discussion by RageMist** [xulihang/ImageTrans-docs#492] [Q&A] [answered]
Hello Sir Xulihang...
Thanks for the Coupon I purchased the Software. I wanted to know how do I add and use EasyOCR in the ImagesTrans application, I couldn't find anything in the Documentation..

**Comment by xulihang** [comment]
Have you installed it?

1. You have to run it as a server using this python script: https://github.com/xulihang/ImageTrans_plugins/blob/master/easyOCR/server/server_easyocr.py
2. Put the plugin files for easyOCR in ImageTrans's plugins folder: https://github.com/xulihang/ImageTrans_plugins/

**Reply by RageMist** [reply]
I don't think its working properly, don't you have any guide for that......and I also saw that other than Azure and Tesseract no other OCR is working......
![image](https://github.com/xulihang/ImageTrans-docs/assets/69992781/a2fb4fc4-7cb9-472d-92d4-aa7a25a0874c)
![Screenshot 2023-09-24 193712](https://github.com/xulihang/ImageTrans-docs/assets/69992781/496143d3-e641-482c-b38a-2f8d594d83d2)

**Reply by RageMist** [reply]
Can you provide any solution for this......

**Comment by xulihang** [comment]
The OCR list is not complete. The OCR plugins are not loaded properly. How did you install ImageTrans? Are you using the package from OneDrive?

What image do you need to translate?

**Reply by RageMist** [reply]
There are more options but it have to be scrolled down...
![image](https://github.com/xulihang/ImageTrans-docs/assets/69992781/f8cee6ae-ca11-4ae6-9af4-566e0bbe7440)

**Reply by RageMist** [reply]
As for the image it a Webtoon.....I'll send you the the first page.....

![001](https://github.com/xulihang/ImageTrans-docs/assets/69992781/b9168c57-1a97-457c-9e4a-01c2c7db1982)
ou the first page.....

**Comment by xulihang** [comment]
If you are using the package from onedrive, restart ImageTrans to see if you can see OCR plugins like WinRT, Google, paddleocr, mangaocr, etc.

**Reply by RageMist** [reply]
Ocrspace doesn't do proper OCR, Tesseract and Azure are 50-50 sometimes they don't detect text sometimes only half of the text, no responce from youdao, abby, clova, google,tencent, winRT,paddleOCR and all baidu response to "null".....and MangaOCR only does japanese and I need Chinese.......

https://github.com/xulihang/ImageTrans-docs/assets/69992781/8a42636d-809a-41f6-903d-bbc4f4ba3d94

**Reply by RageMist** [reply]
So! Sir Xulihang, What should I do now.....

**Reply by xulihang** [reply, marked as answer]
Not all OCRs are available by default as I wrote in the docs: https://imagetrans.readthedocs.io/en/latest/gettingstarted.html#ocr

You have to apply for the API keys to use them, or install and run the engines locally.

Baidu is a service by a Chinese company. It recognizes Chinese accurately. But it seems that it is blocked in India. Youdao and tencent are also Chinese services. They are not applicable in India.

You can try to use Google.

There are two ways to use it.

1. A free option is to use Google Drive. You can use it in ImageTrans directly or run it by yourself: https://github.com/xulihang/ImageTrans-docs/issues/432#issuecomment-1694989036
2. Apply for the API key for Google Vision OCR according to the link in the FAQ page: https://www.basiccat.org/imagetrans/faq/

Of course, there are other options. But they may not be as accurate as Google.

**Comment by xulihang** [comment]
Here is a video using Google Vision OCR:

https://github.com/xulihang/ImageTrans-docs/assets/5462205/e26f52af-665e-4e13-bb76-5630669ccee6

**Comment by xulihang** [comment]
WinRT is Windows 10's built-in OCR. It works great for Chinese as well and it is free as it runs on your device. 

It seems that it is not called correctly. Maybe the language models are lacking. Try to install Chinese as the system's language.

Then you can use it as in the video:


https://github.com/xulihang/ImageTrans-docs/assets/5462205/66cadae5-9cb1-4fb7-b204-12fd09475b5c

**Reply by RageMist** [reply]
I'll give Google a shot, but just in case it don't work for me is there any option to refund or something.....

**Reply by xulihang** [reply]
No problem.

**Reply by RageMist** [reply]
I have another Question if you are still avalible, I want use ChapGPT as Machine Translation because its far betten than the rest but even after add the API key it still doesn't response........
![image](https://github.com/xulihang/ImageTrans-docs/assets/69992781/74aa1bb8-b31c-4a51-80f0-04894e0e1baa)

**Reply by xulihang** [reply]
Check out this blog about ChatGPT: https://www.basiccat.org/ChatGPT-image-translator/

Use `ImageTrans.bat` to start ImageTrans and you can check out its outputs to see why it fails.

**Comment by RageMist** [comment]
I did follow the blog but it still doesn't work for me....I looked at the Google Vision, and its free plan only allows 1000units/month, which might be around 20-25 manhua chapters......The ImagesTrans Application is not bad but maybe for me, it's a bit complex and not suitable.....I hope you can issue a refund. And I'm Sorry for all the trouble....

**Comment by RageMist** [comment]
Sir Xulihang, Do I need to send a Mail with a Refund Request?

**Reply by xulihang** [reply]
Refunded. Please check your account.

**Reply by RageMist** [reply]
I live in India, So I'm sure it will take a few days to credit. but Thank You!
